pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -234,11 +258,36 @@ func (pl *InterPodAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod
 		return framework.Queue, err
 	}
 
+	// When queuing this Pod:
+	// - 1. A new node is added with the pod anti-affinity topologyKey, the pod may become schedulable.
+	// - 2. The original node does not have the pod anti-affinity topologyKey but the modified node does, or vice versa, the pod may become schedulable.
+	// - 3. Both the original and modified nodes have the pod anti-affinity topologyKey and they differ, the pod may become schedulable.
 	for _, term := range antiTerms {
-		if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
-			logger.V(5).Info("a node with matched pod anti-affinity topologyKey was added/updated and it may make pod schedulable",
+		if originalNode == nil {
+			if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
```

Comment:
Rethinking about it, as I found for the test failures, the antiaffinity passes Nodes without `TopologyKey`. So, probably we should return `Queue` at **_any_** Node/Add events, rather than only returning `Queue` for Node/Add with `TopologyKey`?

https://github.com/kubernetes/kubernetes/blob/7b28a115ba04651bc31aa1d7089abbd67ec5c067/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L321-L333

Reply:
What about 2nd scenario then? "The original node does not have the pod anti-affinity topologyKey but the modified node does"
Does adding a `topologyKey` to existing node can make the pod schedulable? I think it won't matter in case of anti-affinity. Am I right?

Reply:
Yes, that's a good catch! 
So, the second scenario should also be changed to include only "The original node had the pod anti-affinity topologyKey but the modified node doesn't" (match -> unmatch case).

Reply:
@googs1025 ↑ can you fix?